### ISSUE TITLE
Support GameController DB

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/vAmiga#843]
I've stumbled across the following repo by chance: https://github.com/mdqinc/SDL_GameControllerDB
It contains game controller descriptions and axis mappings for various devices in a compatible SDL format. 

At the moment, vAmiga (as well as VirtualC64) manage their own internal database for a similar purpose. 

TODO: Check if the database file from the mentioned repo can be utilized instead of my inferior home-brewed variant.

**Comment by dirkwhoffmann** [comment]
The next version will support SDL-compatible mapping schemes:

<img width="758" alt="Bildschirmfoto 2024-12-22 um 10 29 40" src="https://github.com/user-attachments/assets/bebb1be9-e9e7-42b0-b93b-26b04ac09c03" />

The abovementioned [database](https://github.com/mdqinc/SDL_GameControllerDB) has been added, so most devices should be recognized. If a device is not known or if the mapping scheme does not comply with the device, it can be edited manually. The changes will be saved in the Mac's user defaults storage. 

Additionally, I have bumped up the number of simultaneously supported external devices from 2 to 4.
